Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -104,4 +113,84 @@ extension Channel {
 
         return self.pipeline.addHandlers(handlers, position: position).map { multiplexer }
     }
+
+    /// Configures a channel to perform a HTTP/2 secure upgrade.
```

Comment:
This is essentially a copy and paste with `configureHTTP2SecureUpgrade`. Things I changed:
- `h2ChannelConfigurator` and `http1ChannelConfigurator` names to reflect that they are configuring channels
- changed type `(ChannelPipeline) -> EventLoopFuture<Void>` to `(Channel) -> EventLoopFuture<Void>`